src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -98,7 +85,240 @@ private void TypeChecks(TypeSymbol type, BaseFieldDeclarationSyntax fieldSyntax,
+                }
+
+                result |= DeclarationModifiers.Static; // "constants are considered static members"
+            }
+            else
+            {
+                // NOTE: always cascading on a const, so suppress.
+                // NOTE: we're being a bit sneaky here - we're using the containingType rather than this symbol
+                // to determine whether or not unsafe is allowed.  Since this symbol and the containing type are
+                // in the same compilation, it won't make a difference.  We do, however, have to pass the error
+                // location explicitly.
+                containingType.CheckUnsafeModifier(result, errorLocation, diagnostics);
+            }
+
+            return result;
+        }
+
+        internal sealed override void ForceComplete(SourceLocation locationOpt, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var incompletePart = state.NextIncompletePart;
+                switch (incompletePart)
+                {
+                    case CompletionPart.Attributes:
+                        GetAttributes();
+                        break;
+
+                    case CompletionPart.Type:
+                        GetFieldType(ConsList<FieldSymbol>.Empty);
+                        break;
+
+                    case CompletionPart.ConstantValue:
+                        GetConstantValue(ConstantFieldsInProgress.Empty, earlyDecodingWellKnownAttributes: false);
+                        break;
+
+                    case CompletionPart.FixedSize:
+                        int discarded = this.FixedSize;
+                        break;
+
+                    case CompletionPart.None:
+                        return;
+
+                    default:
+                        // any other values are completion parts intended for other kinds of symbols
+                        state.NotePartComplete(CompletionPart.All & ~CompletionPart.FieldSymbolAll);
+                        break;
+                }
+
+                state.SpinWaitComplete(incompletePart, cancellationToken);
+            }
+        }
+
+        internal override NamedTypeSymbol FixedImplementationType(PEModuleBuilder emitModule)
+        {
+            Debug.Assert(!this.IsFixed, "Subclasses representing fixed fields must override");
+            return null;
+        }
+    }
+
+    internal class SourceMemberFieldSymbolFromDeclarator : SourceMemberFieldSymbol
+    {
+        private readonly bool _hasInitializer;
+
+        private TypeSymbol _lazyType;
+
+        // Non-zero if the type of the field has been inferred from the type of its initializer expression
+        // and the errors of binding the initializer have been or are being reported to compilation diagnostics.
+        private int _lazyFieldTypeInferred;
```

Comment:
It is used to suppress duplicate diagnostics from binding a field initializer. Type inference for Out Variables never reports general binding errors, therefore there is nothing to suppress.